### ISSUE TITLE
Demo Guillo & Demonic Strength Conflict

### ISSUE THREAD

**Issue by Raven7576** [Hekili/hekili#2171]
### Before You Begin

- [X] I confirm that I have downloaded the latest version of the addon.
- [X] I am not playing on a private server.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/recommendation) for this issue and was not able to find one.
- [X] I edited the title of this issue (above) so that it describes the issue I am reporting.
- [X] I am reporting an issue with the default priority included with the specialization (imported or edited priorities are not supported).

### WoW Version

Retail (Dragonflight)

### Describe the Issue

Basically, the queue is recommending Guillotine immediately after Demonic Strength. Specifically, the scenario I can recreate is when holding Guillo as a AOE cool down after some movement or a better grouped pack of mobs. 

The biggest piece of this is holding off from casting Guillotine and observing Demonic Strength come off cooldown in the process. 

I managed to record the issue (along with another minor bug with Spellflash): https://youtu.be/wzB8K88JZEU

### How to Reproduce

1. Engage AOE pack or target dummies.
2. Rotate for a few minutes, including casting Demonic Strength and Guillo as needed.
3. Hold off from using Guillo and use other filler. Wait to us Guillo until Demonic Strength comes off cooldown
4. Recommendation changes from Guillo to Demonic Strength
5. DS will change to Guillo for a moment, moving on to Demonbolt after at least 1 GCD.

### Talent Loadout

BoQAAAAAAAAAAAAAAAAAAAAAAggkEJSg0SLCSLJJkWAAAAAQLCSIkIJSSiItkIhAAAAAAA

### Snapshot

https://pastebin.com/4VDYfRE7

### Contact Information

Razør#0394

**Comment by Hekili** [comment]
> The biggest piece of this is holding off from casting Guillotine and observing Demonic Strength come off cooldown in the process.

Can you phrase more clearly what you're seeing and what you'd prefer to see.  It sounds like you don't want to see Guillotine recommended right after Demonic Strength.

Your "How to Reproduce" section is detailed and helpful; I appreciate it.

> 4. Recommendation changes from Guillo to Demonic Strength
> 5. DS will change to Guillo for a moment, moving on to Demonbolt after at least 1 GCD.

So what you mean is that Demonbolt should be the recommendation (with Guillotine unusable) immediately following Demonic Strength?  Or something different?

Raidbots sim links are also very useful for these kinds of reports, to see if the behavior matches the sim profile or not.  Thanks!

**Comment by Raven7576** [comment]
> Can you phrase more clearly what you're seeing and what you'd prefer to see. It sounds like you don't want to see Guillotine recommended right after Demonic Strength.

Basically, Im holding off on using Guillotine as a AOE CD. In keys, I've been holding off on using it on packs of 2 that spread out so that I can wait for the tank to pull more and maximize its effectiveness. In this time, Demonic Strength comes off cooldown too. Hekelli moves DS in front of Guillotine in priority but as soon as I can DS, Guillotine is suggested. This shouldnt be as the two spells can not be cast in the same windows. What I would prefer to see, is at least 1 or GCD's between these spells AND active monitoring of Felstorm so the two arent suggested while the pet is spinning. 

I have "Automatic" toggled so I know its going to fold in basically anything AOE when it can. It feels like a CD just for Mythic+ though. Im far from an expert at the class though so maybe im playing wrong. 

> So what you mean is that Demonbolt should be the recommendation (with Guillotine unusable) immediately following Demonic Strength? Or something different?

Basically im spelling out what Im seeing from this point. 
First, Demonic Strength is recommended, next Gulliotine is reccomended (in error to my view). Half a second or a second later, Demonbolt is being shown by the addon, recovering from its hiccup.

>

My current SimC copy paste from the in-game Addon: https://pastebin.com/MrJLpRt1

Target Dummies- 5 Bosses Quick DPS Results: https://www.raidbots.com/simbot/report/fR1ECazGL5yhGuJvNWXvAT

Honestly dont know which set up would capture this perfectly... Dungeon Slice, Patchwere or Hectic Add Cleave... but I know its most happening during trash packs and the AOE target dummies.

**Comment by Hekili** [comment]
> First, Demonic Strength is recommended, next Gulliotine is reccomended (in error to my view). 

*Why?*

**Comment by Raven7576** [comment]
> > First, Demonic Strength is recommended, next Gulliotine is reccomended (in error to my view).
> 
> _Why?_

The recommendation is immediate. Gulliotine cannot physically be cast until Demonic Strength is done buffing the Fel Guard. There should be a cast in between to account for the filler and down time. Perhaps a demonbolt, shadowbolt or HoG

**Comment by Hekili** [comment]
2.0.8a should address the DS/Guillotine conflict.
